plugin/src/main/java/org/opensearch/ml/action/batch/TransportBatchIngestionAction.java

```diff
@@ -0,0 +1,171 @@
+import org.opensearch.ml.common.transport.batch.MLBatchIngestionInput;
+import org.opensearch.ml.common.transport.batch.MLBatchIngestionRequest;
+import org.opensearch.ml.common.transport.batch.MLBatchIngestionResponse;
+import org.opensearch.ml.engine.MLEngineClassLoader;
+import org.opensearch.ml.engine.ingest.Ingestable;
+import org.opensearch.ml.task.MLTaskManager;
+import org.opensearch.ml.utils.MLExceptionUtils;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class TransportBatchIngestionAction extends HandledTransportAction<ActionRequest, MLBatchIngestionResponse> {
+    private static final String S3_URI_REGEX = "^s3://([a-zA-Z0-9.-]+)(/.*)?$";
+    private static final Pattern S3_URI_PATTERN = Pattern.compile(S3_URI_REGEX);
+    public static final String TYPE = "type";
+    public static final String SOURCE = "source";
+    TransportService transportService;
+    MLTaskManager mlTaskManager;
+    private final Client client;
+
+    @Inject
+    public TransportBatchIngestionAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        MLTaskManager mlTaskManager
+    ) {
+        super(MLBatchIngestionAction.NAME, transportService, actionFilters, MLBatchIngestionRequest::new);
+        this.transportService = transportService;
+        this.client = client;
+        this.mlTaskManager = mlTaskManager;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLBatchIngestionResponse> listener) {
+        MLBatchIngestionRequest mlBatchIngestionRequest = MLBatchIngestionRequest.fromActionRequest(request);
+        MLBatchIngestionInput mlBatchIngestionInput = mlBatchIngestionRequest.getMlBatchIngestionInput();
+        try {
+            validateBatchIngestInput(mlBatchIngestionInput);
+            MLTask mlTask = MLTask
+                .builder()
+                .async(true)
+                .taskType(MLTaskType.BATCH_INGEST)
+                .createTime(Instant.now())
+                .lastUpdateTime(Instant.now())
+                .state(MLTaskState.CREATED)
+                .build();
+
+            mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
+                String taskId = response.getId();
+                try {
+                    mlTask.setTaskId(taskId);
+                    mlTaskManager.add(mlTask);
+                    listener.onResponse(new MLBatchIngestionResponse(taskId, MLTaskType.BATCH_INGEST, MLTaskState.CREATED.name()));
+                    String ingestType = (String) mlBatchIngestionInput.getDataSources().get(TYPE);
+                    Ingestable ingestable = MLEngineClassLoader.initInstance(ingestType.toLowerCase(), client, Client.class);
+                    double successRate = ingestable.ingest(mlBatchIngestionInput);
```

Comment:
Have you tested with a big load ? Will this timeout ?

Reply:
Yes the ML task is returned first and others are handled and processed as a Task behind the scene. I think it's better to use a new thread pool or share the pool with "EXECUTE_THREAD_POOL" or "TRAIN_THREAD_POOL". The TRAIN thread pool is almost idle all the times.